src/dailies.ts

```diff
@@ -845,3 +846,68 @@ function configureShrub(): void {
     "Red Ray"
   );
 }
+
+export function checkBarfQuest(): void {
+  const page = visitUrl("place.php?whichplace=airport_stench&action=airport3_kiosk");
+
+  // If we are on an assignment, try completing and then return after
+  if (page.includes("Current Assignment")) {
+    return completeBarfQuest();
+  }
+
+  const targets = globalOptions.noBarf
+    ? ["Electrical Maintenance"]
+    : ["Track Maintenance", "Electrical Maintenance"]; // In decreasing order of priority
+
+  // Page includes Track/Electrical Maintenance and we aren't on an assignment -> choose assignment
+  const quests = [
+    page.match("(width=250>)(.*?)(value=1>)")?.[2]?.match("(<b>)(.*?)(</b>)")?.[2] ?? "",
+    page.match("(value=1>)(.*?)(value=2>)")?.[2]?.match("(<b>)(.*?)(</b>)")?.[2] ?? "",
+  ];
```

Comment:
There really isn't a need for the filter since there are always 2 quests available. The nullish coalescence is just to ensure that the resultant variable is a string array rather than a string|null array

Reply:
Oh but I guess I should have a check for when we have already completed the quests